File: src/Roguelike/Core/Data/EffectData.cs
namespace Roguelike.Data
{


    public abstract class EffectData
    {


        public string Id { get; set; }


        public string Name { get; set; }


        public string Description { get; set; }


        public int Intensity { get; set; }


        public IntensityType IntensityType { get; set; }


        public int Duration { get; set; } = 1;


        public ApplyType ApplyType { get; set; }


        public DecayType Decay { get; set; }


        public TargetType Target { get; set; }
    }
}

File: python_roguelike/data/effect_data.py
from dataclasses import dataclass, field
from typing import Optional
from .enums import IntensityType, ApplyType, DecayType, TargetType


@dataclass
class EffectData:
    id: str = ""
    name: str = ""
    description: str = ""
    intensity: int = 0
    intensity_type: IntensityType = IntensityType.Flat
    duration: int = 1
    apply_type: ApplyType = ApplyType.RightAway
    decay: DecayType = DecayType.AfterXTURNS
    target: TargetType = TargetType.Self
